test/library/draft/arrow/ArrowDemo.chpl

```diff
@@ -0,0 +1,39 @@
+module ArrowDemo {
+  require "Arrow.chpl";
+  import Arrow;
+  use ArrowDecl;
+  use CPtr;
+
+  proc main(){
+    var arrowInt64Array = Arrow.array([1,2,3,4]);
+    var arrowStringArray = Arrow.array(["foo", "bar", "baz", "some_invalid_value"],[3]);
+    var arrowBooleanArray = Arrow.array([true, false, false, true], [1]);
+
+    var error: GErrorPtr;
+    writeln("Int Array:");
+    printArray(arrowInt64Array);
+    writeln("String Array:");
+    printArray(arrowStringArray);
+    writeln("Int Array:");
+    printArray(arrowBooleanArray);
+    
+    var rcbatch: GArrowRecordBatch;
+    rcbatch = Arrow.recordBatch("field0", arrowInt64Array, "field1", arrowStringArray, "field2", arrowBooleanArray);
+
+    writeln("\nRecord batch from above arrays:");
+    printRecordBatch(rcbatch);
+    
+    writeln("----Parquet Demo-----");
+
+    // Taking the recordBatch above and putting it in a table
+    var table1 = Arrow.table([rcbatch, rcbatch]);
```

Comment:
Mentioned in the issue, but I think the flat list makes more sense, but seems minor and trivial to change if we end up not liking what we decide now

Reply:
I've put in the flat list alternative as well to see if people like how that one looks, but this warrants further discussion.